test/lib/utils.internals.test.js

```diff
@@ -222,3 +222,65 @@ tap.test('#getPropertyValue', t => {
 
   t.end()
 })
+
+tap.test('#parseMessageFormat', t => {
+  const logData = {
+    level: 30,
+    data1: {
+      data2: 'bar'
+    },
+    msg: 'foo'
+  }
+
+  t.test('parseMessageFormat translates if / else statement to found property value', async t => {
+    const log = fastCopy(logData)
+    t.equal(internals.parseMessageFormat('{level} - {if data1.data2}{data1.data2}{end}', log), '{level} - bar')
+  })
+
+  t.test('parseMessageFormat translates if / else statement to found property value and leave unmatched property key untouched', async t => {
+    const log = fastCopy(logData)
+    t.equal(internals.parseMessageFormat('{level} - {if data1.data2}{data1.data2} ({msg}){end}', log), '{level} - bar ({msg})')
+  })
+
+  t.test('parseMessageFormat removes unescaped if statements', async t => {
+    const log = fastCopy(logData)
+    t.equal(internals.parseMessageFormat('{level} - {if data1.data2}{data1.data2}', log), '{level} - {data1.data2}')
+  })
+
+  t.test('parseMessageFormat removes unescaped end statements', async t => {
```

Comment:
What happens if the format string is?:

`{level} - {if foo}something{end}{end}`

(i.e. someone tried to add an unsupported nested condition but failed to add the leading `if`)

Reply:
The additional `{end}` within the if / end statement will be removed. Added test scenario for this.